### PR TITLE
allow selectors containing variants from cbc in noarch recipes

### PR DESCRIPTION
**Pull request by 0xbe7a**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: tests/recipes/noarch_selector_variants/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+requirements:
+  build:
+    - cross-python_{{ target_platform }}    # [build_platform != target_platform]
+    - python                                # [build_platform != target_platform]
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (xhochy) [inline tests/recipes/noarch_selector_variants/meta.yaml:25]: @0xbe7a Can you fix this recipe? I doesn't matter for the test but it should have a python constraint based on `python_leq_310`.

**Review by beckermr — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by tkelman** [comment]
How can we tell whether the linter bot is using a release that includes this change yet? Will that get upgraded by default or does it require manual upgrading of the images / versions the bot uses?

**Comment by pavelzw** [comment]
This will be upgraded by default when there is a new release of this repo (which there was in 3.31.0)

**Comment by tkelman** [comment]
Hm okay, I'm still getting the 

> `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

linter message in botocore, e.g. https://github.com/conda-forge/botocore-feedstock/pull/1495

**Comment by pavelzw** [comment]
At least that's what I've assumed... maybe it works differently 🤔

**Comment by isuruf** [comment]
This PR only supports variants used as booleans. It doesn't support `python_version == "old"`.

**Comment by tkelman** [comment]
Oh didn't realize the code here only works for booleans, we had two simultaneous versions of the variants get implemented over at the feedstock and the text one was what I merged first

**Pull request merged**